### ISSUE TITLE
Show app page preview when hovering over listing on explore page

### ISSUE THREAD

**Issue by Nightmayr** [ubuntu-flutter-community/software#607]
Implement a partial preview of the app page (Screenshot slideshow, description, overall rating) when hovering over the listing/card on the explore page, to provide a better idea of what the app does as well enticing users to find out more about the application by clicking through to the app page. This should also result in reduced number of clicks when discovering applications.
Originally suggested here: https://github.com/ubuntu-flutter-community/software/discussions/601#discussioncomment-4300734

**Comment by madsrh** [comment]
Nice @Nightmayr 👍

**Comment by Feichtmeier** [comment]
I thought more of something like this:

![grafik](https://user-images.githubusercontent.com/15329494/205501731-f1d2615c-47f6-497c-b95e-e101be2d5629.png)

**Comment by madsrh** [comment]
> I thought more of something like this

Okay, yeah we interpreted that very differently 😀 Maybe I took the "a partial preview of the app page" too literary 😆
I really like the show install / open on hover.

**Comment by madsrh** [comment]
@Feichtmeier for the huge card/banner idea to work you would ofc need a short delay, so it doesn't show up as you move across the page. This would also mean that we wouldn't need the loading spinner.

I see it as a peek inside the app page, but in a really compact view. Ofc. there's not room for everything, but a screenshot (slideshow like Steam) and a a few paragraphs of the description.

**Comment by Feichtmeier** [comment]
def. needs a delay, becomes annoying really fast

**Comment by anasereijo** [comment]
@Feichtmeier @madsrh As much as I would love to have the 'install' button displaying on cards, unfortunately that leads us to an issue: what package will it install if that app offers snap and deb? 
As you know, we are 'merging' snaps and debs into the same cards (that's why we now show the icons for packages). 
Let's imagine an app that offers both packages (snaps and debs). If we have the install button within the card, what will it install by default? I explored a few options, but we then agreed this is very complex and needs more thinking, so we left it out from the MVP. Happy to discuss it further with you all, obviously! :)

**Comment by Feichtmeier** [comment]
Good point @anasereijo 
I am also unsure if this really helps
Currently the explore page has those category grids, displaying a grid of banners for the category you select, or a grid of search snaps for the search + the category (if not all, then it's just no category )
Those categories are snap only currently, so maybe we could keep this for those snap only banners? No must have and if we keep it it needs a good transition and a pause, as mads said

**Comment by Nightmayr** [comment]
> Nice @Nightmayr +1 WDYT [justsomeguy5](https://github.com/justsomeguy5) ?
> 
> We still haven't added everything to the "regular" cards - we're still missing things like package format icons, rating and open/install button, but here's a very quick mockup for illustration/discussion. I'll try to do a better and more accurate mockup tomorrow.
> 
> ![image](https://user-images.githubusercontent.com/3986894/205467816-75841e0d-3351-4ba7-9ff4-48dc5db7989d.png)

Nice! This mockup looks really great in my opinion, definitely what I had in mind and I'm in total agreement that it is essentially meant to be a peek into the app page. I think the only info missing in the mockup is the overall rating but as you said that's probably coming in a future release.

 I'm not sure how necessary it is to have the install button present, I think the more valuable aspect is getting more information about the app without going into the page. I think as touched on by @Feichtmeier there should be a decent delay to not annoy the user when scrolling on the page. I think it would be nice if the mockup design is to be pursued, that the resizing of the card animates gracefully as well as the screenshots, description and rating fade in nicely.

**Comment by madsrh** [comment]
> As much as I would love to have the 'install' button displaying on cards, unfortunately that leads us to an issue: what package will it install if that app offers snap and deb?

@anasereijo @Feichtmeier I don't really see the issue here 🙈 What happen when you click the "normal" cards? Do you land on the snap page or the deb page? If the default behavior is to land on the snap page, then then you've already prioritized snaps and then the install button should install the snap.

I've mentioned this before, but I **really** think this out of the box, should be a snap first store, with a setting to make debs the default - here we are only talking about when clicking cards and install buttons ;) 

We could add an icon if that would help
![image](https://user-images.githubusercontent.com/3986894/206014249-68f98294-ff24-4bf5-8cf2-ffea90059348.png)

@Nightmayr 

>  ....the resizing of the card animates gracefully...

I think that will quickly become highly annoying too 😄 A very show fade could work IMO.

**Comment by whiskeyPeak** [comment]
I'm personally not the biggest fan of adding an overlay when hovering over a card. 
I don't remember ever seeing such a feature in any other software store and practically I could see it become quite annoying, if every time the user hovers their mouse above the card, a massive overlay appears with an animation.

The overlay also doesn't add that much more information, to the card (we could maybe add the star count on the base app cards?)

From experience, working with apps that are overly responsive can be somewhat frustrating. Just my 2 cents 🙃

**Comment by madsrh** [comment]
> The overlay also doesn't add that much more information, to the card

It does indeed! Besides the screenshots, you have 8-9 **lines** of description (compared to 8-9 words in the base card) and the install button 😉 
If I was looking for a, let's say first-person-shooter style game, I think finding that from just the icon an a few words would be hard.

> we could maybe add the star count on the base app cards?

This is already the plan 😊 https://github.com/ubuntu-flutter-community/software/discussions/526

> I could see it become quite annoying, if every time the user hovers their mouse above the card, a massive overlay appears with an animation.

I totally agree, but I think this can be done in a way that won't become intrusive. If there's a test branch, we can try and see if it feels too annoying.

**Comment by justsomeguy5** [comment]
Perhaps we take a step back and re-evaluate which requirement the team is looking to solve...

All I originally proposed was maybe make each card a little wider to show a 1-2 sentence description of the app versus the 40-50 char description shown today.  That would eliminate needing to click into so many apps just to learn what it is.  Mahbe show only two columns of cards instead of 3 or 4 columns?

But if the requirement is to enable hovering to show more detail, I again think maybe a paragraph description and maybe a screenshot is enough to allow a user to determine if they want to commit to clicking the card to see snap vs deb, more screenshots, ratings, etc.

> > Nice @Nightmayr +1 WDYT [justsomeguy5](https://github.com/justsomeguy5) ?
> > We still haven't added everything to the "regular" cards - we're still missing things like package format icons, rating and open/install button, but here's a very quick mockup for illustration/discussion. I'll try to do a better and more accurate mockup tomorrow.
> > ![image](https://user-images.githubusercontent.com/3986894/205467816-75841e0d-3351-4ba7-9ff4-48dc5db7989d.png)
> 
> Nice! This mockup looks really great in my opinion, definitely what I had in mind and I'm in total agreement that it is essentially meant to be a peek into the app page. I think the only info missing in the mockup is the overall rating but as you said that's probably coming in a future release.
> 
> I'm not sure how necessary it is to have the install button present, I think the more valuable aspect is getting more information about the app without going into the page. I think as touched on by @Feichtmeier there should be a decent delay to not annoy the user when scrolling on the page. I think it would be nice if the mockup design is to be pursued, that the resizing of the card animates gracefully as well as the screenshots, description and rating fade in nicely.

**Comment by Feichtmeier** [comment]
The more I use this the more I agree that this is super annoying. And this is not changed by what we show on hover
What's the exact problem with clicking one of the app banners/tiles? It's like 1 second to the app, 0.2 seconds back.

**Comment by anasereijo** [comment]
There are a few accessibility points we have to take into consideration If we go with this approach.

First, I am not sure how that would work with **keyboard interactions** as there is no action required (as in, CTA) to trigger that overlay, which probably will just open the overlays as soon as the user hits tab. Then, there is no way to return to the cards view. Same issue for assistive tech users.

**What we can do:** have an explicit CTA (“read more details” or “quick view”) with a close button on the overlay; or hide this functionality from the keyboard navigation. Having the CTA will require a click, so we might just have then the app landing page and not the overlay.

Second, I see an issue with the **change of context**. It’s quite a big overlay that opens without user awareness, hiding most of the content of the page – this can disorient users. I checked Steam's example, and they have more of a tooltip approach. The tooltip isn’t floating around, it’s fixed, and it is much smaller than the approach presented here. I think the current proposal might be annoying when browsing up and down the page with all those big overlays opening on top of the content. 

**What we can do:** maybe we can try something smaller, less intrusive and in a fixed position as Steam’s example (just a paragraph and a screenshot as per @justsomeguy5 suggestions)


We are focusing a lot on the number of clicks, but as @Feichtmeier mentioned, it takes so little time and effort to open an app landing page and come back to the results/explore pages.
As @justsomeguy5 suggested, we can always incorporate longer descriptions within cards.

**Comment by justsomeguy5** [comment]
In a nutshell, this image below is summarizes what I was initially proposing.  Idea is that someone can glean a little more info about an application when you're not hunting for a specific app, but you want to see what the app store contains.

Just an additional sentence or two description, perhaps the first couple lines pulled from the longer application description.

![swstore-after](https://user-images.githubusercontent.com/119681001/206192467-f4854f7c-2b20-415e-baa5-8a86748ee71d.jpg)

**Comment by Feichtmeier** [comment]
I reverted this experiment for now in main until we know what we want :smile:

**Comment by Feichtmeier** [comment]
Any update on this? @anasereijo 

From my side I think this is rather annoying in usage, independt on the animation length

**Comment by madsrh** [comment]
I think we have given this an honest try. Let's close it for now. We can always revisit the idea if someone gets an epiphany.

**Comment by anasereijo** [comment]
> Any update on this? @anasereijo
> 
> From my side I think this is rather annoying in usage, independt on the animation length

Agree! I’d say we can close it for now. We can see how the users respond to the new app and, if we see the need, we can always come back to this and explore further.
